atom/browser/atom_navigation_throttle.cc

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2015 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "atom/browser/atom_navigation_throttle.h"
+
+#include "atom/browser/api/atom_api_web_contents.h"
+#include "content/public/browser/navigation_handle.h"
+
+namespace atom {
+
+AtomNavigationThrottle::AtomNavigationThrottle(
+    content::NavigationHandle* navigation_handle)
+    : content::NavigationThrottle(navigation_handle) {}
+
+AtomNavigationThrottle::~AtomNavigationThrottle() {}
+
+content::NavigationThrottle::ThrottleCheckResult
+AtomNavigationThrottle::WillRedirectRequest() {
+  auto* handle = navigation_handle();
+  auto* contents = handle->GetWebContents();
+  if (!contents) {
+    DCHECK(false);  // This should be unreachable
```

Comment:
`CHECK(false)` then?

Reply:
The reason they are `DCHECK` is to maintain existing behavior in release builds (see the comment chain above).  Although this should be unreachable, in it's current state Electron would `PROCEED` with the navigation.